### PR TITLE
task1

### PR DESCRIPTION
**Pull request by Igor-Gorchakov**

### CODE REVIEW

**Comment by AinuraAliieva** [comment]
**Major points:**

1 Please restructure application by the component reorganization.
>Regarding our app design, we have one header and one footer. So we don't need to re-render it for each screen. For the header, we should change states for inner components only (like breadcrumbs and login).
That's why we could include `<Header />` and `<Footer />` one time into `<App />` and rerender 'body' only (depending on route).
In that case, you don't need `<Page />` at all (and other related components like `<CourseLogo />` etc.). Instead of it, you could have for example `<CoursePageContainer />`, `<LoginPageContainer />` and so on (where you will handle state and other stuff for feature ).

2 Please convert common components to "functions components".
> For common components that don't need Lifecycle methods, we should use a simple function "functions components" (as we don't need additional features).
Details: [Function and Class Components](https://reactjs.org/docs/components-and-props.html#function-and-class-components).

File: src/common/course/CourseForm.jsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleHideModalWindow = this.handleHideModalWindow.bind(this);
+    this.state = {
+        fields: {
+            caption: this.props.caption || "",
+            description: this.props.description || "",
+            date: this.props.date || "",
+            authors: this.props.authors || ""
+        },
```
Comment (AinuraAliieva) [inline src/common/course/CourseForm.jsx]: [JS] could simplify using 'destructuring assignment' with default value for getting params from `this.props`
Something like that:
```
const { caption = "", description = "", date = "", authors = "" } = this.props;

this.state = {
  fields: { caption, description, date, authors },
  errorMessages: [],
...

```

Comment (Igor-Gorchakov) [reply src/common/course/CourseForm.jsx]: Yes, but let's leave it as is, coz this new synthesis may break my mind :)

File: src/common/course/CourseForm.jsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    } else {
+        let validDateRegexp = /^\s*(3[01]|[12][0-9]|0?[1-9])\.(1[012]|0?[1-9])\.((?:19|20)\d{2})\s*$/;
+        if (!validDateRegexp.test(fields["date"])) {
+            errorMessages.push("Type Date field in proper format : dd.mm.yyyy");
+        }
+    }
+    if (fields["authors"] === "") {
+        errorMessages.push("Authors field is required")
+    }
+    this.setState({errorMessages: errorMessages});
```
[the remark this answers is not in the document]
Comment (Igor-Gorchakov) [reply src/common/course/CourseForm.jsx]: Nice point!

File: src/common/course/CourseForm.jsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                        <FormControl componentClass="textarea" placeholder="Description" onChange={this.handleInputChange.bind(this, "description")} value={this.state.fields["description"]} />
+                    </Col>
+                </FormGroup>
+
+                <FormGroup controlId="date">
+                    <Col componentClass={ControlLabel} sm={4}>
+                        Date
+                    </Col>
+                    <Col sm={4}>
+                        <FormControl type="text" placeholder="Date in dd.mm.yyyy format" onChange={this.handleInputChange.bind(this, "date")} value={this.state.fields["date"]}/>
```
[the remark this answers is not in the document]
Comment (Igor-Gorchakov) [reply src/common/course/CourseForm.jsx]: Done

File: src/common/login/LoginForm.jsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  }
+  
+  validateLoginField(loginValue, errorMessages) {
+    if (loginValue === "") {
+      errorMessages["login"] = "Please enter the Login field";
+    } else {
+      let validLoginRegexp = /^[a-zA-Z]+$/i;
+      if (!validLoginRegexp.test(loginValue)) {
+        errorMessages["login"] = "Wrong login : Field «Login» should contain only Latin letters";
+      }
```
[the remark this answers is not in the document]
Comment (Igor-Gorchakov) [reply src/common/login/LoginForm.jsx]: Let's get back here at the end of the source, unfortunately, I don't have time to fix it.

File: src/common/login/LoginForm.jsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      fields: {
+        login: "",
+        password: ""
+      },
+      errorMessages: {}
+    };
+  }
+
+  handleSubmit() {
+    this.validateLoginForm();
```
[the remark this answers is not in the document]
Comment (Igor-Gorchakov) [reply src/common/login/LoginForm.jsx]: Let's get back here at the end of the source, unfortunately, I don't have time to fix it.

**Pull request merged**